src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -37,6 +37,25 @@ public abstract class ExecutionNode
         /// </summary>
         public int? IndexInParentNode { get; protected set; }
 
+        //todo: make public
+        internal IGraphType ResolvedType
+        {
+            get
+            {
+                if (IndexInParentNode.HasValue)
+                {
+                    var t = FieldDefinition.ResolvedType;
+                    if (t is NonNullGraphType nonNullGraphType)
+                        t = nonNullGraphType.ResolvedType;
+                    return ((ListGraphType)t).ResolvedType;
+                }
+                else
+                {
+                    return FieldDefinition?.ResolvedType;
```

Comment:
Why `?.` ?

Reply:
If the `ExecutionNode` is a root node then `FieldDefinition` will return null.